core/sdk-core/src/main/java/software/amazon/awssdk/core/async/SdkPublishers.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2010-2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.core.async;
+
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+
+import org.reactivestreams.Publisher;
+
+import software.amazon.awssdk.annotations.SdkProtectedApi;
+import software.amazon.awssdk.core.internal.async.EnvelopeWrappedSdkPublisher;
+
+/**
+ * Common implementations of {@link SdkPublisher} that are provided for convenience when building asynchronous
+ * interceptors to be used with specific clients.
+ */
+@SdkProtectedApi
+public final class SdkPublishers {
+    private SdkPublishers() {
+    }
+
+    /**
+     * Constructs an {@link SdkPublisher} that wraps a {@link ByteBuffer} publisher and inserts additional content
+     * that wraps the published content like an envelope. This can be used when you want to transform the content of
+     * an asynchronous SDK response by putting it in an envelope.
+     * @param publisher The underlying publisher to wrap the content of.
+     * @param envelopePrefix A string representing the content to be inserted as the head of the containing envelope.
+     * @param envelopeSuffix A string representing the content to be inserted as the tail of containing envelope.
+     * @return An {@link SdkPublisher} that wraps the provided publisher.
+     */
+    public static SdkPublisher<ByteBuffer> envelopeWrappedPublisher(Publisher<ByteBuffer> publisher,
+                                                                    String envelopePrefix,
+                                                                    String envelopeSuffix) {
+        return EnvelopeWrappedSdkPublisher.of(publisher,
+                                              wrap(envelopePrefix),
+                                              wrap(envelopeSuffix),
+                                              SdkPublishers::concat);
+    }
+
+    private static ByteBuffer wrap(String s) {
+        return ByteBuffer.wrap(s.getBytes(StandardCharsets.UTF_8));
```

Comment:
Should we add another parameter to specify Charset instead of assuming it's `UTF_8`? we can make it default to `UTF_8` if not specified

Reply:
Maybe, seems like overkill in this case. I could change the name to wrapUtf8() instead?